draft-ietf-quic-transport.md

```diff
@@ -5858,38 +5858,152 @@ decisions are made independently of client-selected values; a Source Connection
+especially sensitive to having a longer encoding.
+
+Applications to register codepoints in QUIC registries MAY include a codepoint
+as part of the registration.  IANA MUST allocate the selected codepoint unless
+that codepoint is already assigned or the codepoints is the first unallocated
+codepoint in the registry.
+
+
+### Reclaiming Provisional Codepoints
+
+A request might be made to remove an unused provisional registration from the
+registry to reclaim space in a registry, or portion of the registry (such as the
+64-16383 range for codepoints that use variable-length encodings).  This SHOULD
+be done only for the codepoints with the earliest recorded date and entries that
+have been updated less than a year prior SHOULD NOT be reclaimed.
+
+A request to remove a codepoint MUST be reviewed by the designated expert(s).
+The expert(s) MUST attempt to determine whether the codepoint is still in use.
+Experts are advised to contact the listed contacts for the registration, plus as
+wide a set of protocol implementers as possible in order to determine whether
+any use of the codepoint is known.  The expert(s) are advised to allow at least
+four weeks for this search.
+
+If any use of the codepoints is identified by this search or a request to update
+the registration is made, the codepoint MUST NOT be reclaimed.  Instead, the
+date on the registration is updated.  A note might be added for the registration
+recording relevant information that was learned.
+
+If no use of the codepoint was identified and no request was made to update the
+registration, the codepoint MAY be removed from the registry.
+
+This process also applies to requests to change a provisional registration into
+a permanent registration, except that the goal is not to determine whether there
+is no use of the codepoint, but to determine that the registration is an
+accurate representation of any deployed usage.
+
+
+### Permanent Registrations {#iana-permanent}
+
+Permanent registrations in QUIC registries require Specification Required policy
+{{!RFC8126}}, unless otherwise specified.  The designated expert(s) verify that
+a specification exists and is readily accessible.  Expert(s) are encouraged to
+be biased towards approving registrations unless they are abusive, frivolous, or
+actively harmful (not merely aesthetically displeasing, or architecturally
+dubious).  The creation of a registry MAY specify additional constraints on
+permanent registrations.
+
+The creation of a registries MAY identify a range of codepoints where
+registrations are governed by a different registration policy.  For instance,
+the registries for 62-bit codepoints in this document have stricted policies for
+codepoints in the range from 0 to 63.
+
+Any stricter requirements for permanent registrations do not prevent provisional
+registrations for affected codepoints.  For instance, a provisional registration
+for a frame type {{iana-frames}} of 61 could be requested.
```

Comment:
I think this example contradicts these two statements from this PR?

> The "QUIC Frame Types" registry [...] values between 0x00 and 0x3f are assigned using Standards Action or IESG Approval.

> Provisional registrations require Expert Review.

Reply:
You clipped the key piece of the first quote:

> Permanent registrations in this registry are assigned using [...] except for values between 0x00 and 0x3f [...] which are assigned using Standards Action or IESG Approval [...]

Reply:
Ah, thanks for clarifying.

If we decide to allow provisional in [0, 63], then we should add text explaining how the Expert Reviewer should handle those requests, because (unless I missed another sentence) it sounds like they're currently instructed to grant all values except for the first unassigned value. I worry that [0, 63] will fill up pretty quickly if we allow provisional use.

Reply:
I share that worry, but I'm trying to let those go.  The main pressure here is that if things start filling up, then we start ejecting registrations sooner.

Reply:
I fear that ejecting won't always be possible if the provisional code point is widely used. I think disallowing provisional for [0, 63] might be best.

Reply:
We're talking about abuse now.  Those same people could just deploy in this space without creating a registration.